### PR TITLE
feat: honoEnhancer for Vercel, Deno and Netlify

### PR DESCRIPTION
**Pull request by rmarscher**
Follow up to #923. I need to test these three but should be able to...

### CODE REVIEW

**Comment by rmarscher** [comment]
I confirmed Deno works. https://rm-waku-project-93w03wnv4zvw.deno.dev/api/hello

My /about page has a few additions including printing the accept header pulled from the Hono context. https://rm-waku-project-93w03wnv4zvw.deno.dev/about  Ignore the 404 error, that was part of a cloudflare test.

I also deployed the cookies example without issue -
https://rm-waku-cookies-example-bkjhc9rcc5e1.deno.dev/

```
deployctl deploy --project rm-waku-cookies-example --include ./dist --include ./private --entrypoint dist/serve-deno.js
```

Just need to test Netlify and Vercel now...

**Comment by rmarscher** [comment]
I forgot about this PR 😬  I never tested with Vercel and Netlify...

**Comment by rmarscher** [comment]
Netlify works with the hono context - https://67335aaff9d5e608f01f7f2f--waku-project.netlify.app/about

**Comment by rmarscher** [comment]
And Vercel works - https://waku-project2-34ob07khi-rob-marschers-projects.vercel.app/about

OK. This PR is ready to merge.

File: packages/waku/src/lib/plugins/vite-plugin-deploy-deno.ts [hunk 1]
```diff
@@ -18,22 +18,29 @@ import { serverEngine } from 'waku/unstable_hono';
...
+      }
+    } catch {}
+    return c.text('404 Not Found', 404);
+  });
+  return app;
+};
+
+const honoEnhancer =
+  (await configPromise).unstable_honoEnhancer || ((createApp) => createApp);
+const app = honoEnhancer(createApp)(new Hono());
```
Comment (dai-shi) [inline packages/waku/src/lib/plugins/vite-plugin-deploy-deno.ts:43]: Maybe we can do the same for `vite-plugin-deploy-cloudflare.ts` using top-level await.

**Review by dai-shi — APPROVED** [review]
LGTM! Thanks for your contribution!

**Pull request merged**